src/api/app/controllers/staging/staging_projects_controller.rb

```diff
@@ -16,6 +17,22 @@ def show
     @staging_project = @main_project.staging.staging_projects.find_by!(name: params[:name])
   end
 
+  def create
+    authorize @staging_workflow
+    xml_request = (Xmlhash.parse(request.body.read) || {}).with_indifferent_access
+    result = ::Staging::StagingProjectCreator.new(xml_request[:staging_project], @staging_workflow).call
```

Comment:
This is wrong. and you only failed to notice because all your tests have 2 staging projects.

The [] operator returns strings if there is only one element (and should only be used for attributes really)

```
irb(main):010:0> Xmlhash.parse('<workflow><staging_project>example2</staging_project></workflow>')['staging_project']
=> "example2"
```

Prefer elements if you want to ensure an array.

```
irb(main):011:0> Xmlhash.parse('<workflow><staging_project>example2</staging_project></workflow>').elements('staging_project')
=> ["example2"]
```